### ISSUE TITLE
Inference hangs on GPU only when Eager is enabled

### ISSUE THREAD

**Issue by McSlay** [tensorflow/models#8449]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
I made small changes (use of opencv to capture images) to the object_detection_tutorial file.
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow installed from (source or binary): python -m pip install tensorflow
- TensorFlow version (use command below): v2.1.0-rc2-17-ge5bf8de410 2.1.0
- Python version: 3.7
- CUDA/cuDNN version: 10.1 / 7.6.5.32
- GPU model and memory: GTX 960M 2GB or RTX2070 Super 8 GB

**Describe the current behavior**
With GPU and Eager mode enabled, running inference hangs indefinitely after processing a few frames (<15).  
If I then disable Eager mode, it runs fine.

**Describe the expected behavior**
Being able to run inference without hangs with Eager mode enabled. 

**Code to reproduce the issue**

```
import os
import pathlib
import cv2
import numpy as np
import tensorflow as tf

from object_detection.utils import ops as utils_ops

def load_model(model_name):
    base_url = 'http://download.tensorflow.org/models/object_detection/'
    model_file = model_name + '.tar.gz'
    model_dir = tf.keras.utils.get_file(
        fname=model_name,
        origin=base_url + model_file,
        untar=True)
    model_dir = pathlib.Path(model_dir)/"saved_model"
    model = tf.saved_model.load(str(model_dir))
    model = model.signatures['serving_default']
    return model

def run_inference_for_single_image(model, image):
    image = np.asarray(image)
    input_tensor = tf.convert_to_tensor(image)
    input_tensor = input_tensor[tf.newaxis, ...]
    # Run inference
    print("Inference start")
    model(input_tensor)
    print("Inference end")

if "models" in pathlib.Path.cwd().parts:
    while "models" in pathlib.Path.cwd().parts:
        os.chdir('..')

#disable eager mode
#tf.compat.v1.disable_eager_execution()

MODELNAME = 'ssd_mobilenet_v1_coco_2017_11_17'
DETECTION_MODEL = load_model(MODELNAME)

utils_ops.tf = tf.compat.v1
tf.gfile = tf.io.gfile

IMGPATH = PATH_TO_IMAGE
IMAGE = cv2.cv2.imread(IMGPATH)

while True:
    run_inference_for_single_image(DETECTION_MODEL, IMAGE)
```
I ran this from the research\object_detection folder

**Other info / logs**
I am not sure how to support the claim of it being a bug. I tried it on different machines and the code is based on an example. I thought it was because there are no error or warning messages before hanging, it works fine when just using the CPU (with or without Eager mode), it works on GPU without Eager mode and it hangs in a library function.

I never did anything like this before. If i did something wrong or more information is required, please let me know.

**Comment by pkulzc** [comment]
I'm actually surprised to see our saved models can run in TF2. Most of our models are exported to SavedModel in TF1.X environment so they are not guaranteed to work in TF2. You could try reexport them by running export_inference_graph.py, but a better way is to switch to TF1.15 or waiting for our TF2 release (happening soon!).

**Comment by McSlay** [comment]
Hi,
Then I think I asked my question on the wrong page. My question was reffering to this git page https://github.com/tensorflow/models/tree/master/research/object_detection. 

The installation page stated that Tensorflow >=1.12 could be used. For me this included version 2.

The code I used is from an example that was supplied with the code from the git page (research/object_detection/object_detection_tutorial.ipynb.).
When running this with jupyter notebook, the first line is "!pip install -U --pre tensorflow=="2.*"". This installed tensorflow version 2 when I ran it. 

I also tried to run the example code with tensorflow 1.15, but this gave errors. 

Another thing I tried was converting the code to tensorflow 1. This also blocks indefinitely when running the session.run function and would work fine when only using the CPU.
